src/runtime/arc.ts

```diff
@@ -488,29 +460,21 @@ export class Arc implements ArcInterface {
         assert(type.isResolved(), `Can't create handle for unresolved type ${type}`);
 
         const storeId = this.generateID().toString();
-        const volatileKey = recipeHandle.immediateValue ? (
-          Flags.useNewStorageStack ? new VolatileStorageKey(this.id, '').childKeyForHandle(storeId) : 'volatile'
-        ) : undefined;
+        const volatileKey = recipeHandle.immediateValue
+          ? new VolatileStorageKey(this.id, '').childKeyForHandle(storeId)
+          : undefined;
 
         const newStore = await this.createStoreInternal(type, /* name= */ null, storeId,
             recipeHandle.tags, volatileKey, recipeHandle.capabilities, recipeHandle.ttl);
         if (recipeHandle.immediateValue) {
           const particleSpec = recipeHandle.immediateValue;
           const type = recipeHandle.type;
-
-          assert(type instanceof InterfaceType && type.interfaceInfo.particleMatches(particleSpec));
-          const particleClone = particleSpec.clone();
-
-
-          if (Flags.useNewStorageStack) {
-            const proxy = new StorageProxy(this.generateID().toString(), await newStore.activate(), newStore.type, null, recipeHandle.ttl);
-            const handle = unifiedHandleFor({proxy, idGenerator: this.idGenerator, particleId: this.generateID().toString()});
-            // tslint:disable-next-line: no-any
-            await (handle as SingletonHandle<any>).set(particleClone);
+          if (isSingletonInterfaceStore(newStore)) {
+            assert(type instanceof InterfaceType && type.interfaceInfo.particleMatches(particleSpec));
+            const handle: SingletonInterfaceHandle = singletonHandle(await newStore.activate(), this);
```

Comment:
don't you need to pass recipeHandle.ttl?

Reply:
good catch, thanks